src/ScottPlot/plottables/PlottableScatterHighlight.cs

```diff
@@ -0,0 +1,288 @@
+            List<PointF> points = new List<PointF>(xs.Length);
+            for (int i = 0; i < xs.Length; i++)
+                if (!double.IsNaN(xs[i]) && !double.IsNaN(ys[i]))
+                    points.Add(settings.GetPixel(xs[i], ys[i]));
+
+            // draw Y errorbars
+            if (errorY != null)
+            {
+                for (int i = 0; i < points.Count; i++)
+                {
+                    PointF errorBelow = settings.GetPixel(xs[i], ys[i] - errorY[i]);
+                    PointF errorAbove = settings.GetPixel(xs[i], ys[i] + errorY[i]);
+                    float xCenter = errorBelow.X;
+                    float yTop = errorAbove.Y;
+                    float yBot = errorBelow.Y;
+                    settings.gfxData.DrawLine(penLineError, xCenter, yBot, xCenter, yTop);
+                    settings.gfxData.DrawLine(penLineError, xCenter - errorCapSize, yBot, xCenter + errorCapSize, yBot);
+                    settings.gfxData.DrawLine(penLineError, xCenter - errorCapSize, yTop, xCenter + errorCapSize, yTop);
+                }
+            }
+
+            // draw X errorbars
+            if (errorX != null)
+            {
+                for (int i = 0; i < points.Count; i++)
+                {
+                    PointF errorLeft = settings.GetPixel(xs[i] - errorX[i], ys[i]);
+                    PointF errorRight = settings.GetPixel(xs[i] + errorX[i], ys[i]);
+                    float yCenter = errorLeft.Y;
+                    float xLeft = errorLeft.X;
+                    float xRight = errorRight.X;
+                    settings.gfxData.DrawLine(penLineError, xLeft, yCenter, xRight, yCenter);
+                    settings.gfxData.DrawLine(penLineError, xLeft, yCenter - errorCapSize, xLeft, yCenter + errorCapSize);
+                    settings.gfxData.DrawLine(penLineError, xRight, yCenter - errorCapSize, xRight, yCenter + errorCapSize);
+                }
+            }
+
+            // draw the lines connecting points
+            if (penLine.Width > 0 && points.Count > 1)
+            {
+                if (stepDisplay)
+                {
+                    List<PointF> pointsStep = new List<PointF>(points.Count * 2);
+                    for (int i = 0; i < points.Count - 1; i++)
+                    {
+                        pointsStep.Add(points[i]);
+                        pointsStep.Add(new PointF(points[i + 1].X, points[i].Y));
+                    }
+                    pointsStep.Add(points[points.Count - 1]);
+                    settings.gfxData.DrawLines(penLine, pointsStep.ToArray());
+                }
+                else
+                {
+                    settings.gfxData.DrawLines(penLine, points.ToArray());
+                }
+            }
+
+            // draw a marker at each point
+            if ((markerSize > 0) && (markerShape != MarkerShape.none))
+                for (int i = 0; i < points.Count; i++)
+                    if (highlightedIndexes.Contains(i))
```

Comment:
For dozens of highlighted points this `Contains()` work fast. Thousands of `highlilghted` points will significant drop performance because this `Contains()`. 
It's possible to make this check fast if `highlightedindexes` stored sorted. But it's hard to implement and i think don't really need, so:

this is just a note.

Reply:
I decided to sort it, it wasn't difficult